torch/csrc/api/include/torch/data/datasets/chunk.h

```diff
@@ -120,8 +121,9 @@ class BatchDataBuffer {
       AT_ASSERT(
           batch_example_indices &&
           batch_example_indices.value().size() == example_count)
-      BatchRequestType indices = batch_example_indices.value();
+      BatchRequestType& indices = batch_example_indices.value();
       for (size_t i : indices) {
+        AT_ASSERT(i < data_size);
```

Comment:
Please `AT_CHECK`. This might be a user error so it shouldn't trigger an assert that says they should file a bug request.